### ISSUE TITLE
why here use        quat_rotate_inverse to rotate self.d.qvel[0:3] in sim2sim

### ISSUE THREAD

**Issue by nuliguangxiaoyang-crypto** [HBUTHUANGPX/bydmmc_mj_sim2sim#1]
@HBUTHUANGPX ,Hi
thanks for your sharing，
and a question is why here use        quat_rotate_inverse to rotate self.d.qvel[0:3] in sim2sim

 self.single_obs[63:66] = (
            quat_rotate_inverse(
                torch.from_numpy(self.d.qpos[3:7]).unsqueeze(0),
                torch.from_numpy(self.d.qvel[0:3]).unsqueeze(0),
            )
            .detach()
            .cpu()
            .numpy()
        )

But in beyondmimic train code we can not find the corresponding operation
we can only see 
        base_lin_vel = ObsTerm(func=mdp.base_lin_vel, noise=Unoise(n_min=-0.5, n_max=0.5))

**Comment by HBUTHUANGPX** [comment]
Thank you very much for your attention to my work. I also had some doubts when implementing the code at the place you mentioned. The version of IsaacLab I'm using is 2.1.0, and the version of IsaacSim is 4.5.0.

The description of the function ‘mdp.base_ang_vel’ in the [source code](https://github.com/isaac-sim/IsaacLab/blob/v2.1.0/source/isaaclab/isaaclab/envs/mdp/observations.py) is located on lines 47 to 51
```
def base_ang_vel(env: ManagerBasedEnv, asset_cfg: SceneEntityCfg = SceneEntityCfg("robot")) -> torch.Tensor:
    """Root angular velocity in the asset's root frame."""
    # extract the used quantities (to enable type-hinting)
    asset: RigidObject = env.scene[asset_cfg.name]
    return asset.data.root_ang_vel_b
```


It can be seen that the implementation of `root_ang_vel_w` in lines 270-276 and  `root_ang_vel_b` in lines 287-294 of the [IsaacLab source code](https://github.com/isaac-sim/IsaacLab/blob/v2.1.0/source/isaaclab/isaaclab/assets/rigid_object/rigid_object_data.py) is as follows:

```
    @property
    def root_ang_vel_b(self) -> torch.Tensor:
        """Root angular velocity in base world frame. Shape is (num_instances, 3).

        This quantity is the angular velocity of the root rigid body's center of mass frame with respect to the
        rigid body's actor frame.
        """
        return math_utils.quat_rotate_inverse(self.root_link_quat_w, self.root_ang_vel_w)
   @property
    def root_ang_vel_w(self) -> torch.Tensor:
        """Root angular velocity in simulation world frame. Shape is (num_instances, 3).

        This quantity is the angular velocity of the root rigid body's center of mass frame.
        """
        return self.root_state_w[:, 10:13]
```
`self.root_state_w` is defined as follows on lines 112 to 128 of the same file:
```
    @property
    def root_state_w(self):
        """Root state ``[pos, quat, lin_vel, ang_vel]`` in simulation world frame. Shape is (num_instances, 13).

        The position and orientation are of the rigid body's actor frame. Meanwhile, the linear and angular
        velocities are of the rigid body's center of mass frame.
        """

        if self._root_state_w.timestamp < self._sim_timestamp:
            # read data from simulation
            pose = self._root_physx_view.get_transforms().clone()
            pose[:, 3:7] = math_utils.convert_quat(pose[:, 3:7], to="wxyz")
            velocity = self._root_physx_view.get_velocities()
            # set the buffer data and timestamp
            self._root_state_w.data = torch.cat((pose, velocity), dim=-1)
            self._root_state_w.timestamp = self._sim_timestamp
        return self._root_state_w.data
```
In [mujoco 's documentation](https://mujoco.readthedocs.io/en/stable/computation/index.html), it can be seen that qvel' s description of body velocities is also based on the world coordinate system.

<img width="1272" height="554" alt="Image" src="https://github.com/user-attachments/assets/9127cfb2-4ef0-4dfc-9801-980ae8a45e31" />

This is why I performed a rotational transformation on the calculation of base_ang_vel in my open-source code.

However, I need to emphasize that, based on the habits from my previous projects developed with Isaac-gym, base_ang_vel should not require rotational transformation. 

In this work, I tried both the base_ang_vel without rotational transformation and the one with rotational transformation. The former achieved excellent deployment results, running stably for over 2 minutes on the LAFAN1 walk1subject1 dataset, while the latter could only last for a dozen seconds under the same test. However, to ensure consistency with BeyondMimic, I still used the version with rotational transformation in my open-source code.

Tip: This is my first time using Isaac-Sim, and I don’t have a deep understanding of this software framework.
